Automattic/wp-calypso
# MediaModal: move `title` to fields section

### Important

This PR depends on #12032

#### Janitorial PR

This PR is part of the #11093 issue.

---------

This PR moves the item `title` from the top of the dialog to the fields section.

<img src="https://cloud.githubusercontent.com/assets/77539/23815618/a3ecb09a-05c7-11e7-9ad4-1891fd1ad4c9.png" width="500px" />

-----

Fix #11093, #9233
Collides with this PR #11188

------

### Testing

Open an item detail dialog.
_(similar testing that #12032)_

Test edit the title. 
- You could take a look at the network tab paying attention to the requests triggered. It shouldn't trigger a request if the value doesn't change for instance
- Resize the viewport and scroll trying to hide an input and type there. It should be focused.
- Edit close and open the dialog once the changes are done.
